worker.py

```diff
@@ -18,7 +18,8 @@
 
 logging.basicConfig(level=logging.DEBUG)
 
+# Build Queue objects bound to our Redis connection
+queues = [Queue(name, connection=redis_connection) for name in listen]
 
-with Connection(redis_connection):
-    worker = Worker(map(Queue, listen))
-    worker.work()
```

Comment:
I don't know how this works but this would close the connection when done, and the new way won't, right? Maybe that is not needed for this application?